src/nodeSet.ts

```diff
@@ -11,30 +11,32 @@ export const node2 = {
 // TODO(asif): Get correct typings once TS/issue/1863 is resolved.
 export type Node = typeof node1|typeof node2;
 
+let COUNTER = 0;
+
 /**
  * Replacement for WeakSet that checks if a node exists in a set.
  * Optimized for memory usage and performance.
```

Comment:
Do we expect that a node can only be in one set at a time?  If so should we document as such?

Reply:
I don't think thats an assumption anymore. Before, it was because I set it to null; but now that I have a counter, there won't be collisions.

Reply:
My read of this is we have multiple sets, and each set has a unique idx.

But each node, can only map onto one idx at a time.  i.e. node[symbol] is an idx of the set the node is in, but placing it in a new set will override the old one by updating that field.

Reply:
Hmm fair point. I'm going to change this to have new symbol for each nodeset

Reply:
Fixed